types/binary_literal_test.go

```diff
@@ -15,271 +15,269 @@
-		{"0b01101000011001010110110001101100011011110010000001110111011011110111001001101100011001000010000001100110011011110110111100100000011000100110000101110010", []byte("hello world foo bar"), false},
-		{"b'01101000011001010110110001101100011011110010000001110111011011110111001001101100011001000010000001100110011011110110111100100000011000100110000101110010'", []byte("hello world foo bar"), false},
-		{"B'01101000011001010110110001101100011011110010000001110111011011110111001001101100011001000010000001100110011011110110111100100000011000100110000101110010'", []byte("hello world foo bar"), false},
-	}
-	for _, t := range tbl {
-		b, err := ParseBitStr(t.Input)
-		if t.IsError {
-			c.Assert(err, NotNil, Commentf("%#v", t))
-		} else {
-			c.Assert(err, IsNil, Commentf("%#v", t))
-			c.Assert([]byte(b), DeepEquals, t.Expected, Commentf("%#v", t))
+	t.Run("TestParseHexStr", func(t *testing.T) {
+		tbl := []struct {
+			Input    string
+			Expected []byte
+			IsError  bool
+		}{
+			{"x'1'", nil, true},
+			{"x'01'", []byte{0x1}, false},
+			{"X'01'", []byte{0x1}, false},
+			{"0x1", []byte{0x1}, false},
+			{"0x-1", nil, true},
+			{"0X11", nil, true},
+			{"x'01+'", nil, true},
+			{"0x123", []byte{0x01, 0x23}, false},
+			{"0x10", []byte{0x10}, false},
+			{"0x4D7953514C", []byte("MySQL"), false},
+			{"0x4920616D2061206C6F6E672068657820737472696E67", []byte("I am a long hex string"), false},
+			{"x'4920616D2061206C6F6E672068657820737472696E67'", []byte("I am a long hex string"), false},
+			{"X'4920616D2061206C6F6E672068657820737472696E67'", []byte("I am a long hex string"), false},
+			{"x''", []byte{}, false},
 		}
-	}
-	b, err := ParseBitStr("")
-	c.Assert(b, IsNil)
-	c.Assert(err, ErrorMatches, "invalid empty .*")
-}
-
-func (s *testBinaryLiteralSuite) TestParseHexStr(c *C) {
-	defer testleak.AfterTest(c)()
-	tbl := []struct {
-		Input    string
-		Expected []byte
-		IsError  bool
-	}{
-		{"x'1'", nil, true},
-		{"x'01'", []byte{0x1}, false},
-		{"X'01'", []byte{0x1}, false},
-		{"0x1", []byte{0x1}, false},
-		{"0x-1", nil, true},
-		{"0X11", nil, true},
-		{"x'01+'", nil, true},
-		{"0x123", []byte{0x01, 0x23}, false},
-		{"0x10", []byte{0x10}, false},
-		{"0x4D7953514C", []byte("MySQL"), false},
-		{"0x4920616D2061206C6F6E672068657820737472696E67", []byte("I am a long hex string"), false},
-		{"x'4920616D2061206C6F6E672068657820737472696E67'", []byte("I am a long hex string"), false},
-		{"X'4920616D2061206C6F6E672068657820737472696E67'", []byte("I am a long hex string"), false},
-		{"x''", []byte{}, false},
-	}
-	for _, t := range tbl {
-		hex, err := ParseHexStr(t.Input)
-		if t.IsError {
-			c.Assert(err, NotNil, Commentf("%#v", t))
-		} else {
-			c.Assert(err, IsNil, Commentf("%#v", t))
-			c.Assert([]byte(hex), DeepEquals, t.Expected, Commentf("%#v", t))
+		for _, item := range tbl {
+			hex, err := ParseHexStr(item.Input)
+			if item.IsError {
+				require.NotNil(t, err, fmt.Sprintf("%#v", item))
```

Comment:
Could use `NoError(..)` or `Error(..)` for better semantic.

Reply:
Done, PTAL